doc/src/advanced/potential.md

```diff
@@ -0,0 +1,254 @@
+`force` is the negative derivative of the energy with respect to the distance, `r`.
+To be more precise, the vectorial force can readily computed by multiplying the
+result of `force` with the connection vector $\vec{r}$.
+
+The next step is to make our `Potential` usable in Lumol's algortihms to compute
+non-bonded energies and forces.
+Therefore, we have to implement the `PairPotential` trait.
+
+### Implementing `PairPotential`
+
+Let's inspect the [documentation of `PairPotential`](http://lumol.org/lumol/latest/lumol/energy/trait.PairPotential.html).
+
+```rust
+pub trait PairPotential: Potential + BoxClonePair {
+    fn tail_energy(&self, cutoff: f64) -> f64;
+    fn tail_virial(&self, cutoff: f64) -> f64;
+
+    fn virial(&self, r: &Vector3D) -> Matrix3 { ... }
+}
+```
+
+First, we can see that `PairPotential` enforces the implementation of `Potential` which is denoted by `pub trait PairPotential: Potential ...` (we ignore `BoxClonePair` for now).
+That makes sense from a didactive point of view since we said that `PairPotential` is a "specialization" of `Potential`
+and furthermore, we can make use of all functions that we had to implement for `Potential`.
+
+There are three functions in the `PairPotential` trait. The first two functions start with `tail_`.
+These are functions to compute long range or tail corrections.
+Often, we introduce a cutoff distance into our potential beyond which we set the energy to zero.
+Doing so we intoduce an error which we can account for using a tail correction.
+We need two of these corrections, one for the energy, `tail_energy`, and one for the pressure (which uses `tail_virial`
+under the hood). For a beautiful derivation of tail corrections for cut potentials, [see here](https://engineering.ucsb.edu/~shell/che210d/Simulations_of_bulk_phases.pdf).
+
+The third function, `virial`, already has its body implemented -- we don't have to write an implementation for
+our potential.
+
+We will omit the derivation of the formulae for tail corrections here but they are computed by solving these equations
+
+energy: $\int_{r_c}^{\infty} u(r) r^2 \mathrm{d}r$
+
+virial: $\int_{r_c}^{\infty} \frac{\partial u(r)}{\partial r} r^3 \mathrm{d}r$
+
+The implementation looks like so
+
+```rust
+impl PairPotential for Mie {
+    fn tail_energy(&self, cutoff: f64) -> f64 {
+        if self.m <= 3.0 {
+            panic!("No tail correction possible for Mie potential with an exponent lower than 3.0")
+        };
+        let sigma_rc = self.sigma/cutoff;
+        let n_3 = self.n - 3.0;
+        let m_3 = self.m - 3.0;
+        let repulsive = f64::powf(sigma_rc, n_3);
+        let attractive = f64::powf(sigma_rc, m_3);
+        - self.prefac * self.sigma.powi(3) * (repulsive/n_3 - attactive/m_3)
+    }
+
+    fn tail_virial(&self, cutoff: f64) -> f64 {
+        if self.m <= 3.0 {
+            panic!("No tail correction possible for Mie potential with an exponent lower than 3.0")
```

Comment:
In other long ranged potentials, we return from tail_energy/tail_virial instead of panicking.